Trees/BinarySearchTree.hpp

```diff
@@ -0,0 +1,33 @@
+#ifndef BINARY_SEARCH_TREE
+#define BINARY_SEARCH_TREE
+
+struct Node {
+    int value;
+    Node* left_child;
+    Node* right_child;
+};
+
+Node* get_node();
+
+class BinarySearchTree {
+private:
+    Node* root;
+
+    void traversal_inorder_helper(Node* tree);
+    void traversal_preorder_helper(Node* tree);
+    void traversal_postorder_helper(Node* tree);
```

Comment:
It is a recursive function with the current node being passed as an argument. The function call to the traversal methods using objects makes it unnecessary and not possible to pass the current node as an argument to it. This is needed.